### PR TITLE
Integrated add/edit page with Redux

### LINKED ISSUE #100: Integrate event add/edit page with Redux
Right now the event add/edit page (`/src/pages/add-edit/add-edit-page.jsx`) does not use the Redux data store at all.

## What needs to be changed
  * When editing an event, the info should be loaded from the Redux store rather than by making an AJAX request
  * Saving logic (network requests, response handling, etc) should be moved to `/src/data/actions.jsx`

### PR DESCRIPTION
**Pull request by kylecombes**
## Description
Connected the event add/edit with the Redux store to avoid making unnecessary server requests. The add/edit page copies the event data to its own state and then modifies it.

Closes #100.

## Required
Changes must conform to these requirements:
* [x] `yarn test` passes.  All new and existing tests pass.
* [x] `yarn lint` passes. All new code follows the code style of this project.

## Aspirational
We don't yet require these, but they are nice to have:
* [ ] New code is covered by new or existing tests.
* [ ] Changed code is covered by new or existing tests.

### CODE REVIEW

File: src/containers/add-edit-container.js [hunk 1]
```diff
@@ -2,55 +2,55 @@
...
-    }
+  return {
+    general: state.general,
+    eventData: state.events.current,
+    possibleLabels: state.labels.labelList,
+  }
 };
 
-// This function passes functions from /srcs/data/actions.jsx to the React component as props
+// This function passes functions from /src/data/actions.jsx to the React component as props
```
Comment (osteele) [inline src/containers/add-edit-container.js]: It's called `data/actions.js`

File: src/containers/add-edit-container.js [hunk 2]
```diff
@@ -2,55 +2,55 @@
...
-        general: state.general,
-        // events: getVisibleEvents(state.events, state.labels),
-        possibleLabels: state.labels.labelList,
-        markdownGuideVisible: state.addEdit.markdownGuideVisible
-    }
+  return {
+    general: state.general,
+    eventData: state.events.current,
+    possibleLabels: state.labels.labelList,
+  }
```
Comment (osteele) [inline src/containers/add-edit-container.js]: Are we going with [Airbnb semi style](https://github.com/airbnb/javascript#semicolons)? Currently the sources are inconsistent…

File: src/containers/add-edit-container.js (hunk 1 above)
Comment (kylecombes) [reply src/containers/add-edit-container.js]: That it is...

File: src/data/actions.js [hunk 3]
```diff
@@ -200,12 +206,50 @@ export function setViewMode(mode) {
...
+    if (eventData) {
+      dispatch(setCurrentEventData(eventData));
+    } else {
+      // We don't have the data, so request it from the server
+      const url = recId
+        ? `${window.abe_url}/events/${id}/${recId}`
+        : `${window.abe_url}/events/${id}`;
+      axios.get(url)
+        .then(response => dispatch(setCurrentEventData(response.data)))
+        .catch(response => console.log(response)); // TODO Display an error message to the user
```
Comment (osteele) [inline src/data/actions.js]: `console.log` -> `console.error` ?

File: src/containers/add-edit-container.js (hunk 2 above)
Comment (kylecombes) [reply src/containers/add-edit-container.js]: Yeah. I figured I'd do a massive reformatting commit in the near future. Trying to minimize line changes right now, so there's some inconsistency.

File: src/pages/add-edit/add-edit-page.jsx [hunk 4]
```diff
@@ -1,115 +1,71 @@
...
-        let id = eventData.id;
-        let sid = eventData.sid;
-        let rec_id = moment.utc(Number(eventData.rec_id)).toString(); // Reformat as ms since UNIX epoch
-        return id ? id.toString() : (sid + '/' + rec_id);
-    };
+  componentWillReceiveProps(newProps) {
+    if (newProps.eventData) {
+      this.setState({eventData: deepcopy(newProps.eventData)});
+    }
+    // TODO If the event is recurring but newProps.match.params.recId is not defined, copy the data to seriesData
```
[the remark this answers is not in the document]
Comment (kylecombes) [reply src/pages/add-edit/add-edit-page.jsx]: Good to know

**Pull request merged**